pyccel/ast/cwrapper.py

```diff
@@ -0,0 +1,143 @@
+    }
+
+class PyArg_ParseTupleNode(Basic):
+
+    def __init__(self, python_func_args, python_func_kwargs, c_func_args, parse_args, arg_names):
+        if not isinstance(python_func_args, Variable):
+            raise TypeError('Python func args should be a Variable')
+        if not isinstance(python_func_kwargs, Variable):
+            raise TypeError('Python func kwargs should be a Variable')
+        if not isinstance(c_func_args, list) and any(not isinstance(c, Variable) for c in c_func_args):
+            raise TypeError('C func args should be a list of Variables')
+        if not isinstance(parse_args, list) and any(not isinstance(c, Variable) for c in parse_args):
+            raise TypeError('Parse args should be a list of Variables')
+        if not isinstance(arg_names, PyArgKeywords):
+            raise TypeError('Parse args should be a list of Variables')
+
+        if len(parse_args) != len(c_func_args):
+            raise TypeError('There should be the same number of c_func_args and parse_args')
+
+        self._pyarg      = python_func_args
+        self._pykwarg    = python_func_kwargs
+        self._parse_args = parse_args
+        self._arg_names  = arg_names
+        self._flags      = ''
+        i = 0
+        while i < len(c_func_args) and not isinstance(c_func_args[i], ValuedVariable):
+            self._flags += pytype_parse_registry[(c_func_args[i].dtype, c_func_args[i].precision)]
+            i+=1
+        if i < len(c_func_args):
+            self._flags += '|'
+        while i < len(c_func_args):
+            self._flags += pytype_parse_registry[(c_func_args[i].dtype, c_func_args[i].precision)]
+            i+=1
+
+        # Restriction as of python 3.8
+        if any([isinstance(a, Variable) and a.is_kwonly for a in c_func_args]):
+            errors.report('Kwarg only arguments without default values will not raise an error if they are not passed',
+                          symbol=c_func_args, severity='warning')
+
+    @property
+    def pyarg(self):
+        return self._pyarg
+
+    @property
+    def pykwarg(self):
+        return self._pykwarg
+
+    @property
+    def flags(self):
+        return self._flags
+
+    @property
+    def args(self):
+        return self._parse_args
+
+    @property
+    def arg_names(self):
+        return self._arg_names
+
+class PyBuildValueNode(Basic):
+    def __init__(self, result_args = []):
+        self._flags = ''
+        self._result_args = result_args
+        for i in result_args:
+            self._flags += pytype_parse_registry[(i.dtype, i.precision)]
+
+    @property
+    def flags(self):
+        return self._flags
+
+    @property
+    def args(self):
+        return self._result_args
+
+#funccall to change later
+class FuncCall(Basic):
+    def __init__(self, name, args):
+        self._name = name
+        self._args = args
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def args(self):
+        return self._args
```

Comment:
I think this class is confusing matters when we are passing pointers. I have added some logic to the `__init__` method of `FunctionCall` to handle pointers correctly, however to know how to handle them we need to know the signature of the function that we are calling. Unfortunately I think that means we will have to create a `FunctionDef` for every function that we use. We can add them to the bottom of this file with `Py_True` and `Py_False`

Reply:
Once this is fixed then we should have a working pull request. The only remaining work will then be to tidy up the code